crates/oxc_traverse/src/context/scoping.rs

```diff
@@ -452,16 +450,18 @@ impl TraverseScoping {
             return false;
         }
 
-        // Check for unbound references in program with this name
-        !self.symbols.references.iter().any(|reference| {
-            if reference.symbol_id().is_some() {
-                // Skip string comparison on bound references, as they'll also be in `symbols.names`
-                // which already checked above
-                false
-            } else {
-                reference.name().as_str() == name
-            }
-        })
+        true
+
+        // // Check for unbound references in program with this name
+        // !self.symbols.references.iter().any(|reference| {
+        //     if reference.symbol_id().is_some() {
+        //         // Skip string comparison on bound references, as they'll also be in `symbols.names`
+        //         // which already checked above
+        //         false
+        //     } else {
+        //         reference.name().as_str() == name
+        //     }
+        // })
```

Comment:
Is this for compatibility with Babel's logic? Do we have another way to do it? or just delete it. Seems doesn’t have any tests failed. 

@overlookmotel

Reply:
I've removed this code in #4356.